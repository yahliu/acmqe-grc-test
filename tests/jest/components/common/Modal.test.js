/* Copyright (c) 2020 Red Hat, Inc. */
'use strict'

import React from 'react'
import Modal from '../../../../src-web/components/common/Modal'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import * as reducers from '../../../../src-web/reducers'
import { createStore, combineReducers, applyMiddleware, compose } from 'redux'
import thunkMiddleware from 'redux-thunk'
import { BrowserRouter } from 'react-router-dom'
import {
  resourceType,
  removeResourceModalLabels, editResourceModalLabels,
  enableResourceModalLabels, disableResourceModalLabels,
  enforceResourceModalLabels, informResourceModalLabels
} from '../modals/ModalsTestingData'

// Mock YamlEditor since we aren't passing it any actual data
jest.mock('../../../../src-web/components/common/YamlEditor', () => {
  return function mockYamlEditor() {
    return <div data-testid='mockYamlEditor' />
  }
})

const data = {
  kind: 'Policy',
  metadata: {
    name: 'test-policy'
  }
}

describe('Modal component', () => {
  const preloadedState = window.__PRELOADED_STATE__
  const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose
  const middleware = [thunkMiddleware]
  const store = createStore(combineReducers(reducers), preloadedState, composeEnhancers(
    applyMiddleware(...middleware)
  ))
  it('renders RemoveResource as expected', () => {
    const component = mount(
      <BrowserRouter>
        <Modal
          type='resource-remove'
          open={true}
          data={data}
          label={removeResourceModalLabels}
          store={store}
        />
      </BrowserRouter>
    )
    expect(toJson(component.find(Modal))).toMatchSnapshot()
  })
  it('renders EditResource as expected', () => {
    const component = mount(
      <BrowserRouter>
        <Modal
          type='resource-edit'
          open={true}
          data={data}
          label={editResourceModalLabels}
          resourceType={resourceType}
          store={store}
        />
      </BrowserRouter>
    )
    expect(toJson(component.find(Modal))).toMatchSnapshot()
  })
  it('renders EnableResource as expected', () => {
    const component = mount(
      <BrowserRouter>
        <Modal
          type='resource-enable'
          open={true}
          data={data}
          label={enableResourceModalLabels}
          store={store}
        />
      </BrowserRouter>
    )
    expect(toJson(component.find(Modal))).toMatchSnapshot()
  })
  it('renders DisableResource as expected', () => {
    const component = mount(
      <BrowserRouter>
        <Modal
          type='resource-disable'
          open={true}
          data={data}
          label={disableResourceModalLabels}
          store={store}
        />
      </BrowserRouter>
    )
    expect(toJson(component.find(Modal))).toMatchSnapshot()
  })
  it('renders EnforceResource as expected', () => {
    const component = mount(
      <BrowserRouter>
        <Modal
          type='resource-enforce'
          open={true}
          data={data}
          label={enforceResourceModalLabels}
          store={store}
        />
      </BrowserRouter>
    )
    expect(toJson(component.find(Modal))).toMatchSnapshot()
  })
  it('renders InformResource as expected', () => {
    const component = mount(
      <BrowserRouter>
        <Modal
          type='resource-inform'
          open={true}
          data={data}
          label={informResourceModalLabels}
          store={store}
        />
      </BrowserRouter>
    )
    expect(toJson(component.find(Modal))).toMatchSnapshot()
  })
  it('returns null given open=false', () => {
    const component = renderer.create(
      <Modal
        type='resource-remove'
        open={false}
        data={data}
        label={removeResourceModalLabels}
        store={store}
      />
    )
    expect(component.toJSON()).toBeNull()
  })
  it('returns null given no type', () => {
    const component = renderer.create(
      <Modal
        store={store}
      />
    )
    expect(component.toJSON()).toBeNull()
  })
})
