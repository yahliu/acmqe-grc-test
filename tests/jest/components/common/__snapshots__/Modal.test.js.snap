// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component renders DisableResource as expected 1`] = `
<Connect(Modal)
  data={
    Object {
      "kind": "Policy",
      "metadata": Object {
        "name": "test-policy",
      },
    }
  }
  label={
    Object {
      "heading": "modal.disable-hcmcompliance.heading",
      "label": "modal.disable-hcmcompliance.heading",
      "primaryBtn": "modal.disable-hcmcompliance.heading",
    }
  }
  open={true}
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
  type="resource-disable"
>
  <Modal
    data={
      Object {
        "kind": "Policy",
        "metadata": Object {
          "name": "test-policy",
        },
      }
    }
    dispatch={[Function]}
    label={
      Object {
        "heading": "modal.disable-hcmcompliance.heading",
        "label": "modal.disable-hcmcompliance.heading",
        "primaryBtn": "modal.disable-hcmcompliance.heading",
      }
    }
    open={true}
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
    type="resource-disable"
  >
    <ForwardRef
      data={
        Object {
          "kind": "Policy",
          "metadata": Object {
            "name": "test-policy",
          },
        }
      }
      dispatch={[Function]}
      label={
        Object {
          "heading": "modal.disable-hcmcompliance.heading",
          "label": "modal.disable-hcmcompliance.heading",
          "primaryBtn": "modal.disable-hcmcompliance.heading",
        }
      }
      open={true}
      store={
        Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        }
      }
      type="resource-disable"
    >
      <Component
        data={
          Object {
            "kind": "Policy",
            "metadata": Object {
              "name": "test-policy",
            },
          }
        }
        dispatch={[Function]}
        forwardedRef={null}
        label={
          Object {
            "heading": "modal.disable-hcmcompliance.heading",
            "label": "modal.disable-hcmcompliance.heading",
            "primaryBtn": "modal.disable-hcmcompliance.heading",
          }
        }
        open={true}
        store={
          Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          }
        }
        type="resource-disable"
      >
        <InnerLoadable
          data={
            Object {
              "kind": "Policy",
              "metadata": Object {
                "name": "test-policy",
              },
            }
          }
          dispatch={[Function]}
          forwardedRef={null}
          label={
            Object {
              "heading": "modal.disable-hcmcompliance.heading",
              "label": "modal.disable-hcmcompliance.heading",
              "primaryBtn": "modal.disable-hcmcompliance.heading",
            }
          }
          open={true}
          store={
            Object {
              "dispatch": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
              Symbol(observable): [Function],
            }
          }
          type="resource-disable"
        />
      </Component>
    </ForwardRef>
  </Modal>
</Connect(Modal)>
`;

exports[`Modal component renders EditResource as expected 1`] = `
<Connect(Modal)
  data={
    Object {
      "kind": "Policy",
      "metadata": Object {
        "name": "test-policy",
      },
    }
  }
  label={
    Object {
      "heading": "modal.edit-hcmcompliance.heading",
      "label": "modal.edit-hcmcompliance.label",
      "primaryBtn": "modal.button.submit",
    }
  }
  open={true}
  resourceType={
    Object {
      "list": "ALL_POLICIES",
      "name": "HCMCompliance",
    }
  }
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
  type="resource-edit"
>
  <Modal
    data={
      Object {
        "kind": "Policy",
        "metadata": Object {
          "name": "test-policy",
        },
      }
    }
    dispatch={[Function]}
    label={
      Object {
        "heading": "modal.edit-hcmcompliance.heading",
        "label": "modal.edit-hcmcompliance.label",
        "primaryBtn": "modal.button.submit",
      }
    }
    open={true}
    resourceType={
      Object {
        "list": "ALL_POLICIES",
        "name": "HCMCompliance",
      }
    }
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
    type="resource-edit"
  >
    <ForwardRef
      data={
        Object {
          "kind": "Policy",
          "metadata": Object {
            "name": "test-policy",
          },
        }
      }
      dispatch={[Function]}
      label={
        Object {
          "heading": "modal.edit-hcmcompliance.heading",
          "label": "modal.edit-hcmcompliance.label",
          "primaryBtn": "modal.button.submit",
        }
      }
      open={true}
      resourceType={
        Object {
          "list": "ALL_POLICIES",
          "name": "HCMCompliance",
        }
      }
      store={
        Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        }
      }
      type="resource-edit"
    >
      <Component
        data={
          Object {
            "kind": "Policy",
            "metadata": Object {
              "name": "test-policy",
            },
          }
        }
        dispatch={[Function]}
        forwardedRef={null}
        label={
          Object {
            "heading": "modal.edit-hcmcompliance.heading",
            "label": "modal.edit-hcmcompliance.label",
            "primaryBtn": "modal.button.submit",
          }
        }
        open={true}
        resourceType={
          Object {
            "list": "ALL_POLICIES",
            "name": "HCMCompliance",
          }
        }
        store={
          Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          }
        }
        type="resource-edit"
      >
        <InnerLoadable
          data={
            Object {
              "kind": "Policy",
              "metadata": Object {
                "name": "test-policy",
              },
            }
          }
          dispatch={[Function]}
          forwardedRef={null}
          label={
            Object {
              "heading": "modal.edit-hcmcompliance.heading",
              "label": "modal.edit-hcmcompliance.label",
              "primaryBtn": "modal.button.submit",
            }
          }
          open={true}
          resourceType={
            Object {
              "list": "ALL_POLICIES",
              "name": "HCMCompliance",
            }
          }
          store={
            Object {
              "dispatch": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
              Symbol(observable): [Function],
            }
          }
          type="resource-edit"
        />
      </Component>
    </ForwardRef>
  </Modal>
</Connect(Modal)>
`;

exports[`Modal component renders EnableResource as expected 1`] = `
<Connect(Modal)
  data={
    Object {
      "kind": "Policy",
      "metadata": Object {
        "name": "test-policy",
      },
    }
  }
  label={
    Object {
      "heading": "modal.enable-hcmcompliance.heading",
      "label": "modal.enable-hcmcompliance.label",
      "primaryBtn": "modal.enable-hcmcompliance.heading",
    }
  }
  open={true}
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
  type="resource-enable"
>
  <Modal
    data={
      Object {
        "kind": "Policy",
        "metadata": Object {
          "name": "test-policy",
        },
      }
    }
    dispatch={[Function]}
    label={
      Object {
        "heading": "modal.enable-hcmcompliance.heading",
        "label": "modal.enable-hcmcompliance.label",
        "primaryBtn": "modal.enable-hcmcompliance.heading",
      }
    }
    open={true}
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
    type="resource-enable"
  >
    <ForwardRef
      data={
        Object {
          "kind": "Policy",
          "metadata": Object {
            "name": "test-policy",
          },
        }
      }
      dispatch={[Function]}
      label={
        Object {
          "heading": "modal.enable-hcmcompliance.heading",
          "label": "modal.enable-hcmcompliance.label",
          "primaryBtn": "modal.enable-hcmcompliance.heading",
        }
      }
      open={true}
      store={
        Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        }
      }
      type="resource-enable"
    >
      <Component
        data={
          Object {
            "kind": "Policy",
            "metadata": Object {
              "name": "test-policy",
            },
          }
        }
        dispatch={[Function]}
        forwardedRef={null}
        label={
          Object {
            "heading": "modal.enable-hcmcompliance.heading",
            "label": "modal.enable-hcmcompliance.label",
            "primaryBtn": "modal.enable-hcmcompliance.heading",
          }
        }
        open={true}
        store={
          Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          }
        }
        type="resource-enable"
      >
        <InnerLoadable
          data={
            Object {
              "kind": "Policy",
              "metadata": Object {
                "name": "test-policy",
              },
            }
          }
          dispatch={[Function]}
          forwardedRef={null}
          label={
            Object {
              "heading": "modal.enable-hcmcompliance.heading",
              "label": "modal.enable-hcmcompliance.label",
              "primaryBtn": "modal.enable-hcmcompliance.heading",
            }
          }
          open={true}
          store={
            Object {
              "dispatch": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
              Symbol(observable): [Function],
            }
          }
          type="resource-enable"
        />
      </Component>
    </ForwardRef>
  </Modal>
</Connect(Modal)>
`;

exports[`Modal component renders EnforceResource as expected 1`] = `
<Connect(Modal)
  data={
    Object {
      "kind": "Policy",
      "metadata": Object {
        "name": "test-policy",
      },
    }
  }
  label={
    Object {
      "heading": "modal.enforce-hcmcompliance.heading",
      "label": "modal.enforce-hcmcompliance.label",
      "primaryBtn": "modal.enforce-hcmcompliance.heading",
    }
  }
  open={true}
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
  type="resource-enforce"
>
  <Modal
    data={
      Object {
        "kind": "Policy",
        "metadata": Object {
          "name": "test-policy",
        },
      }
    }
    dispatch={[Function]}
    label={
      Object {
        "heading": "modal.enforce-hcmcompliance.heading",
        "label": "modal.enforce-hcmcompliance.label",
        "primaryBtn": "modal.enforce-hcmcompliance.heading",
      }
    }
    open={true}
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
    type="resource-enforce"
  >
    <ForwardRef
      data={
        Object {
          "kind": "Policy",
          "metadata": Object {
            "name": "test-policy",
          },
        }
      }
      dispatch={[Function]}
      label={
        Object {
          "heading": "modal.enforce-hcmcompliance.heading",
          "label": "modal.enforce-hcmcompliance.label",
          "primaryBtn": "modal.enforce-hcmcompliance.heading",
        }
      }
      open={true}
      store={
        Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        }
      }
      type="resource-enforce"
    >
      <Component
        data={
          Object {
            "kind": "Policy",
            "metadata": Object {
              "name": "test-policy",
            },
          }
        }
        dispatch={[Function]}
        forwardedRef={null}
        label={
          Object {
            "heading": "modal.enforce-hcmcompliance.heading",
            "label": "modal.enforce-hcmcompliance.label",
            "primaryBtn": "modal.enforce-hcmcompliance.heading",
          }
        }
        open={true}
        store={
          Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          }
        }
        type="resource-enforce"
      >
        <InnerLoadable
          data={
            Object {
              "kind": "Policy",
              "metadata": Object {
                "name": "test-policy",
              },
            }
          }
          dispatch={[Function]}
          forwardedRef={null}
          label={
            Object {
              "heading": "modal.enforce-hcmcompliance.heading",
              "label": "modal.enforce-hcmcompliance.label",
              "primaryBtn": "modal.enforce-hcmcompliance.heading",
            }
          }
          open={true}
          store={
            Object {
              "dispatch": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
              Symbol(observable): [Function],
            }
          }
          type="resource-enforce"
        />
      </Component>
    </ForwardRef>
  </Modal>
</Connect(Modal)>
`;

exports[`Modal component renders InformResource as expected 1`] = `
<Connect(Modal)
  data={
    Object {
      "kind": "Policy",
      "metadata": Object {
        "name": "test-policy",
      },
    }
  }
  label={
    Object {
      "heading": "modal.inform-hcmcompliance.heading",
      "label": "modal.inform-hcmcompliance.label",
      "primaryBtn": "modal.inform-hcmcompliance.heading",
    }
  }
  open={true}
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
  type="resource-inform"
>
  <Modal
    data={
      Object {
        "kind": "Policy",
        "metadata": Object {
          "name": "test-policy",
        },
      }
    }
    dispatch={[Function]}
    label={
      Object {
        "heading": "modal.inform-hcmcompliance.heading",
        "label": "modal.inform-hcmcompliance.label",
        "primaryBtn": "modal.inform-hcmcompliance.heading",
      }
    }
    open={true}
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
    type="resource-inform"
  >
    <ForwardRef
      data={
        Object {
          "kind": "Policy",
          "metadata": Object {
            "name": "test-policy",
          },
        }
      }
      dispatch={[Function]}
      label={
        Object {
          "heading": "modal.inform-hcmcompliance.heading",
          "label": "modal.inform-hcmcompliance.label",
          "primaryBtn": "modal.inform-hcmcompliance.heading",
        }
      }
      open={true}
      store={
        Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        }
      }
      type="resource-inform"
    >
      <Component
        data={
          Object {
            "kind": "Policy",
            "metadata": Object {
              "name": "test-policy",
            },
          }
        }
        dispatch={[Function]}
        forwardedRef={null}
        label={
          Object {
            "heading": "modal.inform-hcmcompliance.heading",
            "label": "modal.inform-hcmcompliance.label",
            "primaryBtn": "modal.inform-hcmcompliance.heading",
          }
        }
        open={true}
        store={
          Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          }
        }
        type="resource-inform"
      >
        <InnerLoadable
          data={
            Object {
              "kind": "Policy",
              "metadata": Object {
                "name": "test-policy",
              },
            }
          }
          dispatch={[Function]}
          forwardedRef={null}
          label={
            Object {
              "heading": "modal.inform-hcmcompliance.heading",
              "label": "modal.inform-hcmcompliance.label",
              "primaryBtn": "modal.inform-hcmcompliance.heading",
            }
          }
          open={true}
          store={
            Object {
              "dispatch": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
              Symbol(observable): [Function],
            }
          }
          type="resource-inform"
        />
      </Component>
    </ForwardRef>
  </Modal>
</Connect(Modal)>
`;

exports[`Modal component renders RemoveResource as expected 1`] = `
<Connect(Modal)
  data={
    Object {
      "kind": "Policy",
      "metadata": Object {
        "name": "test-policy",
      },
    }
  }
  label={
    Object {
      "heading": "modal.remove-hcmcompliance.heading",
      "label": "modal.remove-hcmcompliance.label",
      "primaryBtn": "modal.remove-hcmcompliance.heading",
    }
  }
  open={true}
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
  type="resource-remove"
>
  <Modal
    data={
      Object {
        "kind": "Policy",
        "metadata": Object {
          "name": "test-policy",
        },
      }
    }
    dispatch={[Function]}
    label={
      Object {
        "heading": "modal.remove-hcmcompliance.heading",
        "label": "modal.remove-hcmcompliance.label",
        "primaryBtn": "modal.remove-hcmcompliance.heading",
      }
    }
    open={true}
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
    type="resource-remove"
  >
    <ForwardRef
      data={
        Object {
          "kind": "Policy",
          "metadata": Object {
            "name": "test-policy",
          },
        }
      }
      dispatch={[Function]}
      label={
        Object {
          "heading": "modal.remove-hcmcompliance.heading",
          "label": "modal.remove-hcmcompliance.label",
          "primaryBtn": "modal.remove-hcmcompliance.heading",
        }
      }
      open={true}
      store={
        Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        }
      }
      type="resource-remove"
    >
      <Component
        data={
          Object {
            "kind": "Policy",
            "metadata": Object {
              "name": "test-policy",
            },
          }
        }
        dispatch={[Function]}
        forwardedRef={null}
        label={
          Object {
            "heading": "modal.remove-hcmcompliance.heading",
            "label": "modal.remove-hcmcompliance.label",
            "primaryBtn": "modal.remove-hcmcompliance.heading",
          }
        }
        open={true}
        store={
          Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          }
        }
        type="resource-remove"
      >
        <InnerLoadable
          data={
            Object {
              "kind": "Policy",
              "metadata": Object {
                "name": "test-policy",
              },
            }
          }
          dispatch={[Function]}
          forwardedRef={null}
          label={
            Object {
              "heading": "modal.remove-hcmcompliance.heading",
              "label": "modal.remove-hcmcompliance.label",
              "primaryBtn": "modal.remove-hcmcompliance.heading",
            }
          }
          open={true}
          store={
            Object {
              "dispatch": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
              Symbol(observable): [Function],
            }
          }
          type="resource-remove"
        />
      </Component>
    </ForwardRef>
  </Modal>
</Connect(Modal)>
`;
